### DISCUSSION TITLE
Emoji characters

### DISCUSSION THREAD

**Discussion by 1stevengrant** [stefanzweifel/sidecar-browsershot#41] [Q&A] [answered]
When I was running Puppeteer locally, I needed to have an additional font package installed.

Is there a way to do that when running server less function?

**Comment by stefanzweifel** [comment, marked as answer]
Good question! 

Honestly, I'm not 100% sure if this would work, but I think this could be do-able with a custom layer.
A quick search brought up this article: https://jurgen.si/posts/lambda-headless-chrome-emojis/

They downloaded a custom font, added them as a layer and referenced them in the Puppeteer script.

sidecar-browsershot currently does have a [`layer`-config](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/config/sidecar-browsershot.php#L18-L23), but that can only be a single value, as we push that into an array [here](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/src/Functions/BrowsershotFunction.php#L69-L79).

To make this easier, we could add a new `layers`-config, so userland can add as many layers as you want.
But one still would need to update the `browser.js`-file and tell Chromium to load the fonts. 🤔 

Or maybe we update sidecar-browsershot to support Emojis out of the box? (I certainly would like that).
Have to think a bit about this.

**Reply by stefanzweifel** [reply]
The layer the package currently uses (https://github.com/shelfio/chrome-aws-lambda-layer) currently uses a deprecated dependency (https://github.com/Sparticuz/chrome-aws-lambda).

The successor project (https://github.com/Sparticuz/chromium) mentions Emoji fonts in their README [here](https://github.com/Sparticuz/chromium#fonts).

Once the layer has been updated to use the new dependency, I think it's save for us to reference and Emoji font in [`browsershot.js`](https://github.com/stefanzweifel/sidecar-browsershot/blob/main/resources/lambda/browsershot.js).
(Might even PR this upstream, as I currently face an issue with not working language overrides with Chromium in the currently used layer)

**Reply by stefanzweifel** [reply]
Good news @1stevengrant: This was easier than I thought (#42).
[v1.5.0](https://github.com/stefanzweifel/sidecar-browsershot/releases/tag/v1.5.0) now supports Emojis out of the box by using Google's Noto Color Emoji font. 🎉
